reference/configuration/framework.rst

```diff
@@ -536,14 +557,20 @@ To configure a Redis cache pool with a default lifetime of 1 hour, do the follow
     .. code-block:: php
 
         // config/packages/framework.php
-        use Symfony\Config\FrameworkConfig;
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
 
-        return static function (FrameworkConfig $framework): void {
-            $framework->cache()
-                ->pool('cache.mycache')
-                    ->adapters(['cache.adapter.redis'])
-                    ->defaultLifetime(3600);
-        };
+        return App::config([
+            'framework' => [
+                'cache' => [
+                    'pools' => [
+                        'cache.mycache' => [
+                            'adapter' => 'cache.adapter.redis',
```

Comment:
This is a service id, but we could update the config to accept a `Reference`:

```diff
-                           'adapter' => 'cache.adapter.redis',
+                           'adapter' => service('cache.adapter.redis'),
```

And maybe an inline `Definition`:
```diff
-                           'adapter' => 'cache.adapter.redis',
+                           'adapter' => inline_service(RedisAdapter::class)->...,
```

Reply:
Not convinced we should diverge from what's possible in yaml.